cmd/root.go

```diff
@@ -355,11 +358,12 @@ func ExecuteContext(ctx context.Context, rootCmd *cobra.Command, clients *shared
 				clients.IO.PrintDebug(ctx, "Exiting with cancel exit code.")
 				os.Exit(int(iostreams.ExitCancel))
 			}()
+		// Received cancelled context, so send an interrupt signal
 		case <-ctx.Done():
-			// No interrupt signal sent, command executed to completion
-			// FIXME - `.Done()` channel is triggered by `cancel()` and not a successfully completion
+			clients.IO.PrintDebug(ctx, "Got a cancelled context, sending interrupt signal")
+			interruptChan <- os.Interrupt
```

Comment:
note: 🍞 🧈 Here's the bread & butter of the PR. 

The `<-ctx.Done()` channel is triggered when the context is cancelled with `cancel()`. Previously, this was not handled. So unless the `<-completedChan` or `<-interruptChan` were triggered, the CLI would hang - although this was very unlikely to happen.

Now, when the context is cancelled, we send an `<-interruptChan`. This seems reasonable, since the context will be cancelled for an unexpected reason, so the interrupt signal matches well (I think).

FWIW, the only time that the context was cancelled was:

1. Part of the cleanup from handling the interrupt signal
2. Deferred function when `ExecuteContext(...)` returns

In both of these cases, either `completedChan` or `interruptChan` should have been triggered.